agent/agent.go

```diff
@@ -0,0 +1,76 @@
+package agent
+
+import (
+	"fmt"
+	"net"
+	"os"
+
+	"github.com/PlakarKorp/plakar/appcontext"
+)
+
+type Daemon struct {
+	socketPath string
+	listener   net.Listener
+	ctx        *appcontext.AppContext
+}
+
+func NewDaemon(ctx *appcontext.AppContext, network string, address string) (*Daemon, error) {
+	if network != "unix" {
+		return nil, fmt.Errorf("unsupported network: %s", network)
+	}
+
+	d := &Daemon{
+		socketPath: address,
+		ctx:        ctx,
+	}
+
+	if _, err := os.Stat(d.socketPath); err == nil {
```

Comment:
actually i forgot a small nitpick: I'm not sure about this extra stat, are you afraid of unlinking an existing file?

If I compare this with the "usual" `control.c` file, we just do an unlink (and error if it fails with something different than `ENOENT`), bind the socket and chmod it.

Reply:
yeah I don't want to unlink it if it exists and there's still an agent listening to it, I want to unlink it only if it exists but is a left-over from a previous run that crashed or was aborted.

Reply:
ok i see